Get pip update make targets running again

maeve-fpf opened the pull request:
This cleans up the Makefile targets that update {dev-,}requirements.txt, and:

- runs them with a python 3.5 image to match current (legacy? :)) prod python version
- does not install all of the dev requirements just to be able to run `pip-compile` in the container

Without this, for reasons I don't entirely understand, `pip install` installs from dev-requirements.txt fine, but attempting to run any `pip-compile` step fails; logs from running manually in a container here: https://gist.github.com/maeve-fpf/a55248dcbf8420b0542c114f92beca1a

I suspect it may be related to the fact that if you pulled an official python docker image a long time ago, it'd have pip 19, but if you pull one now, it's on pip 20.

maeve-fpf commented:
Testing note: please make sure you've pulled any python docker images that you have locally. `make compile-pip-dependencies` should no longer crash.

chigby commented on Makefile:
```diff
@@ -6,6 +6,7 @@ UID := $(shell id -u)
 GIT_REV := $(shell git rev-parse HEAD | cut -c1-10)
 GIT_BR := $(shell git rev-parse --abbrev-ref HEAD)
 STN_IMAGE := quay.io/freedomofpress/securethenews
+PY_IMAGE := python:3.5-slim
```
Yes! This is great.

chigby commented on Makefile:
```diff
@@ -37,31 +38,31 @@ compile-pip-dependencies: ## Uses pip-compile to update requirements.txt
-    pip install --require-hashes -r /code/dev-requirements.txt && \
-		pip-compile --generate-hashes --no-header --output-file /code/requirements.txt /code/requirements.in && \
-		pip-compile --generate-hashes --no-header --allow-unsafe --output-file /code/dev-requirements.txt /code/dev-requirements.in'
+		pip install pip-tools && \
```
I wonder if it would be a good idea here to pin the version of `pip-tools`? I can also see though waiting to do that until we can also pin the version of `pip`, since it seems like one of the underlying causes of this command failing is an incompatibility between these two libraries.

As a point of reference, the way it's being pinned in one of the other repos is [with a separate requirements file just for pip tools](https://github.com/freedomofpress/tracker/blob/master/Makefile#L49), which also lets us use `--require-hashes` when installing. I don't really love that solution, as it proliferates the number of requirements files and probably ought to have _another_ command to update the pip tools version, and it kind of seems like a lot of commands and files but it does pin the version and require the hash.

It also seems perfectly reasonable to just `pip install pip-tools` like you have here. I don't know.. what do you think?

chigby commented on Makefile:
```diff
@@ -37,31 +38,31 @@ compile-pip-dependencies: ## Uses pip-compile to update requirements.txt
 
 .PHONY: pip-dev-update
-update-pip-dev: ## Uses pip-compile to update dev-requirements.txt for upgrading a specific package
+pip-dev-update: ## Uses pip-compile to update dev-requirements.txt for upgrading a specific package
```
This is also great! It fixes something that was totally my fault and continued to annoy me every time I noticed it. Thank you!

maeve-fpf commented:
Updated this to take pip-tools out of `dev-requirements.in`, ran a hash update, fixed another probable typo.

My thinking is that since we're not doing anything to keep `python:3.5-slim` at a particular hash, it will continue to be upgraded, and if we pin `pip-tools` to a particular version, we could get in the same situation again where they are not compatible. Is that reasonable?

chigby commented:
That makes a lot of sense to me. Thanks for recompiling the requirements (and fixing that other error), I'm going to merge this, now.
